$OpenBSD: patch-qtbase_mkspecs_features_qt_build_paths_prf,v 1.1.1.1 2014/12/08 20:33:41 zhuk Exp $
Unbreak building of QtScript.

The roots of problem are likely in QtWebKit, but it's really hard to find them.
--- qtbase/mkspecs/features/qt_build_paths.prf.ports.orig	Tue Nov 11 16:41:30 2014
+++ qtbase/mkspecs/features/qt_build_paths.prf	Tue Nov 11 20:51:42 2014
@@ -13,6 +13,15 @@
 isEmpty(_QMAKE_CONF_): error("Project has no top-level .qmake.conf file.")
 MODULE_BASE_INDIR = $$dirname(_QMAKE_CONF_)
 MODULE_BASE_OUTDIR = $$shadowed($$MODULE_BASE_INDIR)
+isEmpty(MODULE_BASE_OUTDIR) {
+    REAL_BUILD_ROOT = $$(REAL_BUILD_ROOT)
+    isEmpty(REAL_BUILD_ROOT) {
+        error("MODULE_BASE_OUTDIR failed to be translated from $${MODULE_BASE_INDIR}")
+    } else {
+        MODULE_BASE_OUTDIR=$${REAL_BUILD_ROOT}/$$basename(MODULE_BASE_INDIR)
+        message("MODULE_BASE_OUTDIR failed to be translated from $${MODULE_BASE_INDIR}, failing back to $${MODULE_BASE_OUTDIR}")
+    }
+}
 !isEmpty(MODULE_BASE_DIR): MODULE_SYNCQT_DIR = $$MODULE_BASE_DIR    # compat for webkit
 isEmpty(MODULE_SYNCQT_DIR): MODULE_SYNCQT_DIR = $$MODULE_BASE_INDIR
 isEmpty(MODULE_QMAKE_OUTDIR): MODULE_QMAKE_OUTDIR = $$MODULE_BASE_OUTDIR
